sdk/eventgrid/azure-eventgrid/azure/eventgrid/_consumer.py

```diff
@@ -20,31 +20,57 @@
 
 _LOGGER = logging.getLogger(__name__)
 
-from ._models import DeserializedEvent
+from ._models import CloudEvent, EventGridEvent
+from ._helpers import _deserialize_data
 
 class EventGridConsumer(object):
     """
     A consumer responsible for deserializing event handler messages, to allow for access to strongly typed Event objects.
     """
-
-    def deserialize_event(self, event, **kwargs):
-        # type: (Union[str, dict, bytes]) -> models.DeserializedEvent
-        """Single event following CloudEvent/EventGridEvent schema will be parsed and returned as DeserializedEvent.
-        :param event: The event to be deserialized.
-        :type event: Union[str, dict, bytes]
-        :rtype: models.DeserializedEvent
+    def decode_cloud_event(self, cloud_event, **kwargs):
+        # type: (Union[str, dict, bytes], Any) -> CloudEvent
+        """Single event following CloudEvent schema will be parsed and returned as Deserialized Event.
+        :param cloud_event: The event to be deserialized.
+        :type cloud_event: Union[str, dict, bytes]
+        :rtype: CloudEvent
 
         :raise: :class:`ValueError`, when events do not follow CloudEvent or EventGridEvent schema.
         """
         encode = kwargs.pop('encoding', 'utf-8')
         try:
-            if isinstance(event, six.binary_type):
-                event = json.loads(event.decode(encode))
-            elif isinstance(event, six.string_types):
-                event = json.loads(event)
-            return DeserializedEvent(event)
+            if isinstance(cloud_event, six.binary_type):
+                cloud_event = json.loads(cloud_event.decode(encode))
+            elif isinstance(cloud_event, six.string_types):
+                cloud_event = json.loads(cloud_event)
+            deserialized_event = CloudEvent.deserialize(cloud_event)
```

Comment:
The bits north of this appear to be the same in both cases.  I can't imagine the low-level decoding/json bits changing much between the two; feel free to disagree (as it'd be the major blocker to what I'm about to suggest) but in absence of a risk in that, why not move this into the shared mixin I mention in my other comment?  Let me know if this ask is unclear and I can pseudocode some handwaving.